### PR TITLE
Separate badge type buttons into different groups

### PR DESCRIPTION
**Pull request by EliAndrewC**
We now have two rows of badge selection:
- The first row is for selecting between a single or a group registration.  This row ONLY appears on the main prereg form.
- The second row is for selecting badge level (regular, tshirt, supporter, etc).  It shows up on all the prereg forms (main prereg form, badge confirmation page, badge transfer page, and group badge claiming page).

As discussed in #1217 we don't show any of badge levels lower than what's already selected.

I made a couple of other fixes/improvements, which I'll explain with inline code comments.

### CODE REVIEW

File: uber/custom_tags.py [hunk 1]
```diff
@@ -105,6 +105,14 @@ def join_and(xs):
         return ', '.join(xs)
```
Comment (EliAndrewC) [inline uber/custom_tags.py]: On the main prereg form, we want to show the overall price of a thing, e.g.
- Add a Tshirt: $70
- Supporter: $100

However, on the other pages such as when claiming an already-paid-for group badge or filling out the badge transfer form, we want to just list the price added since otherwise people might think they're being charged the base badge price again:
- Add a Tshirt: +$20
- Supporter: +$50

This function looks at which page we're on and chooses between the two.

File: uber/site_sections/preregistration.py [hunk 2]
```diff
@@ -438,7 +438,10 @@ def transfer_badge(self, session, message='', **params):
                 if attendee.full_name in c.BANNED_ATTENDEES:
                     send_banned_email(attendee)
```
Comment (EliAndrewC) [inline uber/site_sections/preregistration.py]: This is a semi-related fix.  When transferring a badge, if someone decides to kick in extra, we now immediately redirect them to the Attendee Donation form.

File: uber/static/theme/prereg.css [hunk 3]
```diff
@@ -15,10 +15,6 @@ body {
     background-repeat: repeat-x repeat-y;
 }
```
Comment (EliAndrewC) [inline uber/static/theme/prereg.css]: Because of the new form rows, we've adjusted our padding rules slightly.  Things pretty much look the same now as they did before.

File: uber/templates/base.html [hunk 4]
```diff
@@ -170,7 +170,6 @@
                 <br/><br/>
             </div>
         {% endblock %}
```
Comment (EliAndrewC) [inline uber/templates/base.html]: This is an unrelated fix.  We're already using ``toastr`` to display these messages, so there's no reason to have a separate div.

File: uber/templates/preregistration/confirm.html [hunk 5]
```diff
@@ -41,6 +41,10 @@
     </div>
 {% endif %}
```
Comment (EliAndrewC) [inline uber/templates/preregistration/confirm.html]: The way the code works is that the Javascript in ``preregbase.html`` now looks at whether or not divs exist with ids ``reg-types`` and/or ``badge-types`` and adds the form elements if they're present.

File: uber/templates/preregistration/form.html [hunk 6]
```diff
@@ -4,8 +4,7 @@
 
 <script>
     $(function () {
```
Comment (EliAndrewC) [inline uber/templates/preregistration/form.html]: We no longer need the spacer row because of the changes to our padding rules.

File: uber/templates/preregistration/form.html [hunk 7]
```diff
@@ -15,83 +14,6 @@
 {% if attendee.is_dealer %}
     <span class="text-center"><h2>Dealer Application</h2></span>
 {% else %}
```
Comment (EliAndrewC) [inline uber/templates/preregistration/form.html]: This all moved into ``preregbase.html`` and it's pretty much the same except for the second row of options.

File: uber/templates/preregistration/form.html [hunk 8]
```diff
@@ -119,8 +41,12 @@
         </div>
     </div>
 
-    <div class="form-group" id="badge_types">
-        <label for="badge" class="col-sm-2 control-label">&nbsp;</label>
```
Comment (EliAndrewC) [inline uber/templates/preregistration/form.html]: Here's where the two rows are defined.

Side note: I spent about 15 minutes trying to get the labels to vertically center.  Vertical centering is notoriously the hardest thing to do in CSS and I completely failed to get it working.  Not a huge deal, but after this is merged, if someone with more CSS experience wants to take a stab at it I'd be grateful :)

File: uber/templates/preregistration/preregbase.html [hunk 9]
```diff
@@ -1,35 +1,135 @@
 {% extends "base.html" %}
-{% block backlink %}
-
-
-{% endblock %}
+{% block backlink %}{% endblock %}
 {% block header %}
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: Unrelated change; we were previously doing a check every 90 seconds.  I suspect this check sometimes fails and that's why some people complain they can't fill out the form before their browser refreshes.  To mitigate the issue somewhat I updated the check to fire every 15 minutes instead.

File: uber/templates/preregistration/preregbase.html [hunk 10]
```diff
@@ -1,35 +1,135 @@
...
+                }
+            });
+        };
+        setTimeout(checkCap, PREREG_CHECK_INTERVAL);
 
-{% if c.COLLECT_FULL_ADDRESS %}
-$(document).ready(function() {
-    $("#international").hide()
-    setInternational()
-});
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: This was just a style change; we're following more standard indentation and quoting conventions now.

File: uber/templates/preregistration/preregbase.html [hunk 11]
```diff
@@ -1,35 +1,135 @@
...
+                }
+            }
+            $(function() {
+                $('#international').hide()
+                setInternational()
+            });
+        {% endif %}
 
-function setInternational() {
-    countryName = $("input[name='country']").val();
```
Comment (EliAndrewC) [inline uber/templates/preregistration/preregbase.html]: Below is the new code that was moved here from ``form.html``.  There are now two global Javascript variables:
- ``REG_TYPES`` controls the first row and currently just determines whether it's a single or a group registration
- ``BADGE_TYPES`` controls the donation level

Both of the above can be overridden in plugins as before.  The main difference is that these two variables each contains several fields:
- ``row`` is the id which is where the options will get appended
- ``selector`` is the class name of the items in that row
- ``options`` is the array of objects which define each option

The main difference between before a

File: uber/templates/regform.html [hunk 12]
```diff
@@ -32,9 +32,13 @@
         noCellphoneClicked();
         $.field('no_cellphone').on('click', noCellphoneClicked);
     });
-</script>
 
-<div id="form-begin-spacer" class="form-group"></div>
```
Comment (EliAndrewC) [inline uber/templates/regform.html]: This is where we remove the badge levels which are lower than the amount which already exists.

File: uber/templates/static_views/stafferComps.html [hunk 13]
```diff
@@ -10,7 +10,7 @@ <h3 align="center"> What You Get For Volunteering </h3>
 <ul>
     <li> Working at least 6 hours gets you a {{ c.EVENT_NAME }} t-shirt. <br/><br/> </li>
     <li> Working at least 12 hours gets you food in addition to the shirt.  We have professional chefs preparing 3 meals per day throughout the event. <br/><br/> </li>
```
Comment (EliAndrewC) [inline uber/templates/static_views/stafferComps.html]: This is the fix for https://github.com/magfest/ubersystem/issues/1215

**Comment by EliAndrewC** [comment]
Another PR was opened for the MagClassic repo at https://github.com/magfest/magclassic/pull/2

**Comment by kitsuta** [comment]
Looks good to me. I'll merge this and close my PR that replicates most of this functionality.
